src/main/java/uk/gov/pay/api/service/PaymentSearchService.java

```diff
@@ -0,0 +1,172 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.gov.pay.api.app.config.PublicApiConfig;
+import uk.gov.pay.api.auth.Account;
+import uk.gov.pay.api.exception.BadRequestException;
+import uk.gov.pay.api.exception.SearchChargesException;
+import uk.gov.pay.api.model.PaymentError;
+import uk.gov.pay.api.model.PaymentForSearchResult;
+import uk.gov.pay.api.model.PaymentSearchResponse;
+
+import javax.inject.Inject;
+import javax.ws.rs.ProcessingException;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.UriBuilder;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static java.util.Arrays.asList;
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+import static org.apache.http.HttpStatus.SC_OK;
+import static uk.gov.pay.api.model.TokenPaymentType.DIRECT_DEBIT;
+import static uk.gov.pay.api.resources.PaymentsResource.CARD_BRAND_KEY;
+import static uk.gov.pay.api.resources.PaymentsResource.DISPLAY_SIZE;
+import static uk.gov.pay.api.resources.PaymentsResource.EMAIL_KEY;
+import static uk.gov.pay.api.resources.PaymentsResource.FROM_DATE_KEY;
+import static uk.gov.pay.api.resources.PaymentsResource.PAGE;
+import static uk.gov.pay.api.resources.PaymentsResource.REFERENCE_KEY;
+import static uk.gov.pay.api.resources.PaymentsResource.STATE_KEY;
+import static uk.gov.pay.api.resources.PaymentsResource.TO_DATE_KEY;
+import static uk.gov.pay.api.resources.PaymentsResource.TRANSACTION_TYPE_KEY;
+import static uk.gov.pay.api.resources.PaymentsResource.TRANSACTION_TYPE_KEY_VALUE;
+import static uk.gov.pay.api.validation.PaymentSearchValidator.validateSearchParameters;
+
+public class PaymentSearchService {
+
+    private static final Logger logger = LoggerFactory.getLogger(PaymentSearchService.class);
+    private static final String PAYMENTS_PATH = "/v1/payments";
+    private final ConnectorUriGenerator connectorUriGenerator;
+    private final Client client;
+    private final ObjectMapper objectMapper;
+    private final String baseUrl;
+    private final PaymentUriGenerator paymentUriGenerator;
+
+    @Inject
+    public PaymentSearchService(Client client,
+                                PublicApiConfig configuration,
+                                ConnectorUriGenerator connectorUriGenerator,
+                                PaymentUriGenerator paymentUriGenerator,
+                                ObjectMapper objectMapper) {
+        this.client = client;
+        this.connectorUriGenerator = connectorUriGenerator;
+        this.paymentUriGenerator = paymentUriGenerator;
+        this.objectMapper = objectMapper;
+        this.baseUrl = configuration.getBaseUrl();
+    }
+    
+    public HalRepresentation doSearch(Account account, String reference, String email, String state, String cardBrand,
```

Comment:
I would split this massive method into smaller ones. Too many things happening inside on method

Reply:
Next iteration perhaps, now I'm just moving everything into a service